src/synced_commands.cpp

```diff
@@ -247,10 +247,7 @@ SYNCED_COMMAND_HANDLER_FUNCTION(disband, child, spectator)
 
 	current_team.recall_list().erase_if_matches_id(unit_id);
```

Comment:
This PR feels like the wrong amount of change; if it should be done then more should be done too.

Given that the reason for saying that an assert can be used is that `dismissed_unit` was checked earlier, why doesn't this use the result of the previous search instead of searching again? However, that leads into asking why the previous search doesn't use `extract_if_matches_id()` instead.

Reply:
The current code spereates the vaildity check from the execution, if we for exaple wanted to add a "pre disband" wml event that would be harder if the code was interwined in the way you suggested,